legal-api/src/legal_api/models/legal_entity.py

```diff
@@ -563,9 +563,11 @@ def find_by_operating_name(cls, operating_name: str = None):
         return None
 
     @classmethod
-    def find_by_identifier(cls, identifier: str = None):
+    def find_by_identifier(cls, identifier: str = None, skip_identifier_validation: bool = False):
```

Comment:
Added this `skip_identifier_validation` flag to be used in amalgamations when fetching an expro business as the business identifier for expros does not pass the `validate_identifier` check.

Reply:
the name says skip identifier validation but the below implementation is opposite, it validates if skip_identifier_validation is True

Reply:
@argush3 do we need to validate identifier in this function?

Reply:
Maybe not.  

@thorwolpert do you have any opinions on this?  I think you added this piece of code.  

We already validate the identifier when we set the identifier property on LE.  

https://github.com/bcgov/lear/blob/95579b862cf27093747a9a4df7432b299c5b03fe/legal-api/src/legal_api/models/legal_entity.py#L245-L251

Or were you wanting to block invalid searches so it never hits the db?

Reply:
I think we can just pull in the code from `main` to get things working and just remove `skip_identifier_validation` for now.

@JazzarKarim is working on syncing `main` with `feature-legal-name` for the BE right now so anything missing pieces should be pulled in soon too.

Reply:
Okay sounds good, yeah the main branch changes means we no longer need `skip_identifier_validation`. I can remove the flag. 

Should I copy the relevant main branch changes over in this PR or just wait for the rebase? The tests will pass without the main branch changes.

(Not sure what happened to my previous GH comment on this thread, but I was able to confirm the main branch changes work without this flag through testing).